static/js/modules/line-chart.js

```diff
@@ -0,0 +1,243 @@
+'use strict';
+
+/* global module */
+var $ = require('jquery');
+var _ = require('underscore');
+var d3 = require('d3');
+var numeral = require('numeral');
+var helpers = require('./helpers');
+
+var parseM = d3.time.format('%b');
+var parseMY = d3.time.format('%b %Y');
+var parseMDY = d3.time.format('%b %e, %Y');
+
+var bisectDate = d3.bisector(function(d) { return d.date; }).left;
+
+function LineChart(selector, snapshot, data, index) {
+  this.element = d3.select(selector);
+  this.data = data;
+  this.index = index;
+
+  this.margin = {top: 10, right: 20, bottom: 30, left: 50};
+  this.baseWidth = $(selector).width();
+  this.baseHeight = this.baseWidth * 0.5;
+  this.height = this.baseHeight - this.margin.top - this.margin.bottom;
+  this.width = this.baseWidth - this.margin.left - this.margin.right;
+
+  this.chart = this.buildChart();
+
+  this.$snapshot = $(snapshot);
+  this.$prev = this.$snapshot.find('.js-snapshot-prev');
+  this.$next = this.$snapshot.find('.js-snapshot-next');
+
+  if (helpers.isMediumScreen()) {
+    this.$snapshot.height(this.baseHeight - this.margin.bottom);
+  }
+
+  this.element.on('mousemove', this.handleMouseMove.bind(this));
+  this.$prev.on('click', this.goToPreviousMonth.bind(this));
+  this.$next.on('click', this.goToNextMonth.bind(this));
+}
+
+LineChart.prototype.buildChart = function() {
+  var data = this.data;
+  var entityTotals = {};
+  var max = 0;
+
+  data.forEach(function(d) {
+    d.date = new Date(d.date);
+  });
+
+  // Transform the data
+  var entityNames = d3.keys(data[0])
+    .filter(function(key) {
+      return key !== 'date';
+    });
+
+  // Create different objects for each entity type
+  entityNames.forEach(function(type) {
+    var totals = data.map(function(d) {
+      return {
+        date: d.date,
+        amount: d[type]
+      };
+    });
+
+    var entityMax = d3.max(totals, function(d) {
+      return d.amount;
+    });
+
+    max = entityMax > max ? entityMax : max;
+    entityTotals[type] = totals;
+  });
+
+  var minYear = d3.min(data, function(d) {
+    return d.date.getFullYear();
+  });
+
+  var maxYear = minYear + 1;
+
+  // Build the scales
+  var x = d3.time.scale()
+    // .domain(d3.extent(data, function(d) {
```

Comment:
Oops, I meant to remove that. It was defining the min and max of the x-scale by the min and max dates in the data set, but I think it's better to show the full two years. I'll remove it.